cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/common/DelimitedStringsRecordFormat.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.template.etl.common;
+
+import co.cask.cdap.api.common.Bytes;
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.data.format.UnexpectedFormatException;
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.api.data.schema.UnsupportedTypeException;
+import com.google.common.base.Charsets;
+import com.google.common.base.Splitter;
+import com.google.common.collect.Lists;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.Charset;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Kafka event record format that interprets the message payload as string of delimited fields.
+ *
+ * <p>
+ * The delimiter can be explicitly set through the "delimiter" setting, and the character set can also be set through
+ * the "charset" setting. By default, the format will use a schema of one field, where the field is an array of strings.
+ * The schema can be set to a schema of fields, with the i'th field corresponding to the i'th value in the delimited
+ * text. Fields can also be parsed as scalar types - boolean, integer, long, double, float, bytes, and string.
+ * In addition, the very last field can be an array of strings.
+ * </p>
+ */
+public class DelimitedStringsRecordFormat extends KafkaEventRecord<StructuredRecord> {
```

Comment:
Looks like there is going to be a lot of code duplication between streams and kafka record format.